### DISCUSSION TITLE
Current state of Standard Grading in Lazer

### DISCUSSION THREAD

**Discussion by Tunnelbliick** [ppy/osu#26671] [Q&A] [answered]
Hey, I wanted to discuss the whole S rank situation in osu!lazer. It's a bit of a mess, from what I can see, and I'm not quite sure where it's heading.

So, there's this Pull Request #26630 that aims to reintroduce the 0 miss requirement for S ranks. But if you check out Issue #4184 [with this resolution](https://github.com/ppy/osu/issues/4184#issuecomment-1724954327), it suggests that S ranks should require ≥95% accuracy and a Full Combo (FC). Personally, I'm really enjoying the current S rank requirements with the 95% accuracy, and I have a feeling I'm not the only one. If you look back at previous changelogs, you'll find comments both in favor and against the current system.

Some players are all for the 95% accuracy requirement: (i didnt cherry pick a changelog this is just the most recent one i remember posts about it beeing below)

https://osu.ppy.sh/comments/2980621
https://osu.ppy.sh/comments/2981104
https://osu.ppy.sh/comments/2981159

While others aren't so keen on it, as seen in the same thread:

https://osu.ppy.sh/comments/2980486

So, what's the plan here? Bringing back the 0 miss requirement won't necessarily mean that S ranks will be ≥95% FC; it'll just mean you need ≥95% accuracy with 0 misses, which isn't the same thing as FC because of sliderbreaks (unless you count them as a miss aswell).

The whole situation with S ranks seems a bit all over the place right now, and some clarity on whats planned right now would be really helpful. In my opinion, keeping the 95% accuracy requirement is a good call. Instead, it could be considered adding a special icon on the leaderboard to show if someone has achieved a Full Combo. That way, we can distinguish between plays with and without FCs more easily, reintroducing the 0 miss requirement wont acchive this if it is the same way as in stable.

**Comment by bdach** [comment, marked as answer]
S rank will require 95% accuracy and no full misses, i.e. hitcircle misses. Sliderbreaks and dropped sliders don't count. There will be a separate indicator for full combo.

**Reply by Tunnelbliick** [reply]
Thanks for the clarification.

I can already see a bigger talk about this erupting with the next release.

**Reply by bdach** [reply]
can't please everybody /shrug

**Comment by Speykious** [comment]
I've looked at other discussion and issue threads. Peppy pointed me to quite a few discussions involving grading, but none were about discussing this latest revert or trying to find a compromise. I feel like outside of making a new discussion, this one is the most apt for me to give feedback. I'm gonna somewhat echo what I said in the osu! dev Discord.

I come forward from [this tweet from peppy](https://twitter.com/ppy/status/1749415464743444611) showing that the S grade will now require FC on the standard ruleset. This is in my opinion the wrong direction. I understand that this is just so the team can move forward for now as it is up for future discussion - which is exactly why I'm giving feedback. With lazer having grades only based on accuracy instead of combo, osu! would be going into a really nice direction where accuracy is more important and rewarding, which is what makes sense for a _rhythm_ game. Having combo be that much of a factor is to go back even harder to the combo game that people tend to complain about. Last time I saw this kind of discussion on Reddit, people were rather massively upvoting in favor of the new grades being independent from the combo. There were a few people being against S being littered through the leaderboard because now it feels less special, but overall the one-miss punishment is too much of a problem for that to be more important.

So yeah, both std grading systems have a flaw:
- S requiring FC, well, requires no miss. If you have 99% one miss, you lose that grade, yet in the eyes of players they totally deserve it.
- S not requiring FC feels too easy to obtain compared to stable. Seeing a 95% 30 miss score graded S on the leaderboard feels wrong.

Note that in both systems, a 95% score with 30 misses and a 99% score with 1 miss are in the same ballpark: either S, which is overrated for the former, or A, which is underrated for the latter.

As a bit of a bias, I started playing osu! on the lazer client in April 2021. (It evolved a whole lot since then!) And recently, I went back to the stable client to try to get my rank to be the same as on lazer in the hopes of participating to the 5 digit world cup. I got to have more of a feel as to what you have to do to climb the ladder.

I've had an experience a long time ago where I told someone that I could do 93% on 7* diffs, and got told it was bad accuracy. From my point of view as a lazer player, it wasn't, since it was just 2% away from an S grade. But from a stable player, that's totally understandable and now I see why. To have any significant score on stable, you need a really high combo and the accuracy naturally follows. If you get an FC with 93% accuracy, that's really bad accuracy. Get 95% and it's still not really good.

## Proposition

Of course, I'm sure many people before have suggested different grading systems, but we have to take into account the current grading system (especially in stable). Adding more grades can fix some problems, but would confuse the large existing osu! player base.
This to me all points to an obvious solution: raising the accuracy bar of all grades in the grading system.
One possibility, for instance:
| Grade | % | Addendum |
| --- | --- | --- |
| SS | 100% | (obviously also means FC) |
| S | ≥ 98% | no FC required |
| A | ≥ 92% | no FC required |
| B | ≥ 85% | no FC required |
| C | ≥ 75% | no FC required |

Keeping the same grades but raising the bar higher, I think, would lead to the least amount of confusion. Not only that, but it would set up new lazer players (like me back then) to better expectations. Also now C nicely fits as 3/4ths of the total.
If people still want combo to be somewhat taken into account, a variation of [the idea of having grades based on score](https://github.com/ppy/osu/discussions/26051) (as a ratio compared to the max score to take multipliers into account) can be applied directly without changing the grading, and that way we have S for ≥ 98% *score*. I'm not sure how well the score scales compared to the accuracy, but if it's drastically different the values would have to be changed I guess. That said, if that is to be decided, I'd hope it would be after making acc be something like 70% of the score.

This is more similar to some other rhythm games, for example Arcaea having: below A < 92, A >= 92 < 95, AA >= 95 < 98, EX >= 98 < 99, and EX+ >= 99. Here notably, getting an EX is way more rewarding than getting an AA grade. That said, comparing to other grading systems is hard since the mechanics of the game and especially how the score is calculated based on the performance plays a major role in the experience.

Note that now, a 93% and 97% score get the same grade (A), but this is much better than having 95% FC and 99% FC being the same, while 99% one-miss is below both. Hypothetically, in an even further future, one could imagine something like A+ being 95-97.99%, but again my proposal is just to raise the bar at least at first.

I might add more thoughts to this comment later but this is the gist.

**Reply by LocalGuard039** [reply]
how is this even equal when we have harder judgement and if not perfect hit then only 33%acc receive? Players who new to lazer would never see the light. And isn't aa equals to S and EX+ is a triple S we never have? These grades must compare to below and upper.

**Reply by Speykious** [reply]
It's not about being equal. One system overrates and the other underrates, so this is a good middle ground. If players who are new to lazer never see the light with this system, then players who started on stable won't either by the same logic. In practice, that's not what happens, nor does it happen in other rhythm games with that kind of accuracy bar, so this is perfectly fine.

Arcaea's AA is equivalent to osu!'s current S without FC requirement, yes. It being S is precisely the overrating problem I'm trying to address with this proposition.

**Reply by LocalGuard039** [reply]
Actually both lazer and stable never see the light is not the same issue. Because grade in stable is mentioned by wiki and not the game, and grade system isn't straightforward by checking accuracy but instead check 300, with a doublestandard that a miss could value 10% 300 in any map any length, thus even system give a grade, we still need to calculate ourself to see what exact place we are. Beginner of Osu stable probably just playing mindlessly.

Lazer on otherhand is a straightforward one,  what acc gives what grade, even a first time would see it clear, and you can see any grade below A are already nerfed comparing to their "with miss" counterpart from stable. And what difference between A and S? 90%with miss and 90%no miss versus 90% and 95% flat change.     Stable S worth a minimum 93.17%acc, and Lazer have slider accuracy too. Honestly I don't even think a miss value equal to 2%acc especially higher the acc harder it to maintain, and to the extra gap compare to stable.

And one more thing, this game has a large quit rate both for 1 miss retry or actually quit, before one actually have skill more than expert. Might thanks to some history issues or some community weird maps. That is, the map quality isn't stable as well. And you guys do cheer for a 95% play if that map is wild, right? So.....

**Reply by Speykious** [reply]
I'm sorry, I think I'm having a lot of trouble trying to understand your points...
- You're talking about the grade being mentioned in the wiki and not the game, but... That's just a documentation issue? And if lazer is more straightforward on this, it's because of the design of the result screen, right? My proposal wouldn't make it worse, it would just change the values, I don't see any downside here.
- You're talking about having to calculate yourself to see what place you are. But osu! tells you what place you are, why calculate it yourself?
- I don't understand what you meant at all in your second paragraph...
- Not sure what you mean about the large quit rate, but I do retry a lot of times on stable compared to lazer. That's because combo is more important, a problem I'm trying to address here again as I said.

**Reply by LocalGuard039** [reply]
1, Grade mention is a design problem indeed, but the value thing is a standardize issue that you and I have a different opinion about how much gap between grades is legit, that I think 98 is too high especially I said the judgement is strict with a only 33%acc second judgement thing, so as for other grades rise.

2, the calculate myself thing is that I won't know where my goal is unless I really would read wiki, issue isn't on calculate itself. To my experience the player introduce me to this was to have 5 days playtime but he don't know a thing about how grades are given.

3,that said in stable the difference is miss or not, as 1%meh normally won't be an issue if one can hit 90%great. In Lazer this difference is already turned to flat 5%acc gap. I think it's quite enough to differ A and S.

4. quit beatmap or quit the game entirely. The combo is too important even for now, like if miss one or two in inappropriate place would ruin everything, might not directly related to this grading system but I think combo stuff could even use more nerf, like pp no longer depends on combo or depend part of it, and if pp becomes easier then nobody would talk if S should be fc or even higher acc then. And always playing the same map isn't a thing that should be encourage for, what enough is enough.
And for quitting the game entirely, well, not directly related to this too. It's just that we all download some terrible map, and newcomer downloaded them they more likely to quit, problem isn't quit, is those unstable quality. 
That is, those maps own different quality, but now   those proposed change, or step back consider comfortable map as guideline instead of harder map. That's why I asked if you guys do cheer for 95% plays if map is wild.

I'm not much a talker in real life nor English master, sorry for those confusion.

**Reply by Speykious** [reply]
I see.

1. 98% is indeed quite high, but this is the goal here. S is *supposed* to be that high in the eyes of players. That's why they find it unfair that 95% 30 miss gets an S grade on lazer, it feels less special. And it's not like you'd get B otherwise, A is still a somewhat good grade, which you can still achieve with a lot of non-greats giving only 33% acc.
2. I still don't understand your point 2, what's the "goal" you're talking about here?
3. *"I think it's quite enough to differ A and S."* - Well, I thing it's quite enough too, but people (especially stable players) disagree, so my best bet is to propose this compromise, because I definitely don't want any grade to require FC.
4. *"I think combo stuff could even use more nerf"* - I addressed that in my message in one of my later edits. At one point peppy did a poll on Twitter about whether acc or combo should be more important, 70% of people voted for acc. So I'm hopeful that we'll see this being changed in that direction in the future. Best scenario would be that the current percentages get reversed: 70% acc, 30% combo.
    - *"It's just that we all download some terrible map, and newcomer downloaded them they more likely to quit, problem isn't quit, is those unstable quality."* - I don't think that's a problem. There's a whole beatmap ranking system for that purpose and I don't think its quality check has loosened in any way. osu!lazer gives you the ability to download maps directly from the client for free, and by default it's only ranked maps of featured artists, you have to go out of your way to download bad maps.
    - I don't think I'm considering any particular style of map here. The grading system is global to the entire standard ruleset, it shouldn't depend on specific mapping styles. I think that's more of a concern for pp calculations.

**Reply by LocalGuard039** [reply]
1, we don't only compare grades to their upper and latter, we also compare with other gamemodes. This 98%S, the fruit picking guy next door shares the same, so that would be completely unreasonable.

2, if I don't know a thing about grading system, then what I can do is get as accurate as I can, but not particularly for S because I don't know how to achieve that. That's what I mean no goal.

3,I don't want compromise at all, even giving them a glowy S I still don't want give away the credit I think well earned. Bowing down would just give them wrong idea, may even lead to more step back further.

4, yes, not directly related.

Everything is clear now... I suppose.

**Reply by Speykious** [reply]
1. ... You're saying that two rulesets (std and catch) *sharing the same acc for S* is unreasonable? Why? lazer's old std grades were the same as mania and you were fine with that. Are you maybe under the impression that 95% S on std is the same as 98% S on catch? Because this is completely arbitrary and I really don't think it's a factor in grading. If you can get 98% way more easily on catch than you can on std relatively speaking, then that's a separate problem on catch.
2. Now you'd know how to achieve S: get above 98% accuracy. That's the compromise I'm proposing.
3. Between my proposition and going back to the stable gradings, both downgrade some S scores into A. It's either that or you overrate. People *clearly* think that giving S on a 95% 30 misses is overrating, so I don't think S>=98% would be a step down at all.

**Reply by Tunnelbliick** [reply]
Has anyone thought about binding grade cutoffs to OD.

Id imagine that probably wont happen since it adds needless complexity to grades, but just a thought.

This would allow for having more dynamic grades, more relaxed grading for easier beginner maps with lower od / wider hit times and harder harshed grading for higher difficulty maps with higher od and more strict hit timings.

Issue i see with that approach tho is that most maps nowadays use od9+

**Reply by Speykious** [reply]
I don't think that's necessary. Having wider hit times means you're already more likely to have a higher accuracy on your play, meaning also a better grade. Lowering the grades themselves at the same time seems like overkill.

**Reply by LocalGuard039** [reply]
1, Oh I can't compare standard with catch fruit, it's arbitrary but you can use arecca as example to raise your point, good for you then.

2, If have to compromise then why the hell do we join Lazer before pp update? Not just for a note lock, this grade standardization also offered a great help.

3, People think 30 miss 95%S is not worthy. Yes, yes, yes, I would say if they can hit 30+ miss and their hit on fellow objects after miss won't be influenced, that's quite a skill. And that is a quite difficult map too or there won't be enough objects to recover accuracy.
Just set this straight, why would we argue a thing for its edge case? These unnecessary challenge could bury edge case you said, as well as many talented.

OD shouldn't determine grade too. Actually I even think multiple standards for each map about great, ok and meh is already bad, but this is a bigger history issue though, exist in map not the game, those wide map pool.
People get higher grade in normal than hard is quite common, but if grades are same then hard one surely better than normal one, they are not the same thing.

**Reply by Speykious** [reply]
1. I never said you *can't* compare std with catch, I said your comparison is weird because you seem to think that them *having the same acc* is unreasonable, which to me doesn't make sense. Unlike your comparison with catch to make your point, my proposition would put osu!'s grades *closer* to Arcaea and not further- not that it matters that much though, since as I said comparing grade systems across different rhythm games with such widely different mechanics is hard, and hardly helps. I just put it there as a reference. I'm not having any kind of double standard here.
2. I have no clue what you're asking here...
3. We argue a thing for this edge case because players think it's important. As a player, I agree with them and think these two cases would benefit from having a separate grade. That's it. If a map is really hard and even top players never get 98%, then so be it. When eventually some player gets an S ≥98% on it, that'll be seen as a great achievement for passing an *accuracy* threshold, whereas today that kind of achievement is an FC instead.

**Reply by LocalGuard039** [reply]
You are just trying to prove your point.

1, You can make grade closer to arecca and "no further", then how about I even raise a point that Osu grade closer to Muse Dash and no further? I say that MD has only 2 rails so notes stack together, just like in Osu we use 2 finger against circle swarm, they just use their aim hand to click too, and they even give you S only for 90% accuracy, how about that? Maybe we both raise ridiculous idea so you would see better.

2, stable is quite playable, then why do we join Lazer in the first place, that's because they made some sensible changes, how is this hard to understand?

3, You can play this game for a quite a long time, and you think 95% isn't suitable for you, and because you already played a lot you can get this accuracy easier or you are patient enough to try again and again.
Hardcore players can afford all efforts to gain these stuff, but not normal players. They want their credit as long as they think they paid enough, not the effort high as 98% accuracy.
I would add again, you think 98% is legit only because you don't consider what difficulty it is, gaining a good result in easier difficulty completely make sense , you think "why they share same grade as mine", but it isn't even the same map, or what if it is? Real top players would swarm those edge case quickly, only you are here mad about that.

Players in any other rhythm games is like that too, and can easily achieve high result in lower difficulty, but they don't have stuff like pp, so they always aim for harder maps(in the meanwhile revenge to maps they played before, majority for harder though), that's why they won't complain about grading system. Only us Osu players complain about how someone share same place with me, heh.

**Reply by Speykious** [reply]
1. Look, at this point I'm just gonna completely disregard your point about comparing other grade systems. It is indeed getting ridiculous, but you failed every single time to understand that I was only putting it as a reference and always ignored that I was saying it didn't matter that much. "We just also aim" is also a completely separate kind of ridiculous that has nothing to do with these grading system comparisons, please have some good faith here. It feels like we've been going around in circles for the past 10 messages and there's nothing productive coming out of this conversation.
2. Having a focus on accuracy, coupled with the removal of note lock and addition of slider head accuracy, can only be beneficial. S being at 98% is totally fine in that regard. I don't know why you think it's a problem.
3. Again, I was perfectly fine with 95%. Stable players weren't. If this change wasn't being reverted, I wouldn't even need to speak up. If stable never existed and we got lazer first, I wouldn't need to speak up either. Stable players constitute the largest portion of the user base. There's a legacy of players that has to be taken into account. New players count too, such as myself back then - again, I started on lazer. But here you're basically completely disregarding the opinion of stable players, you're saying they're just wrong for thinking that and that they should just consider casual players. I think this is the wrong mindset to begin with. I'm trying to consider every side here.

Complaining about grading and how the game rewards players for their performance is bound to happen in a rhythm game- it doesn't even need to be competitive. In Arcaea for example, while there are no complaints about people's ranks and grades that I can see (because they've been good since the very beginning and based on accuracy, as a matter of fact), there are very frequent complaints about how X chart shouldn't be a 10 because it's too easy, or it should be a 10+ because it's too hard. It can also sometimes be a mystery when you lose some PTT, even though the formula is known. It seems like you're trying to take other rhythm games into account but... Not osu! stable players themselves, for some reason.

**Reply by LocalGuard039** [reply]
Wow, you keep saying 98% is what you want, but fine with 95%, I see that from very beginning. But when come to this change your only thought is compromise but no intention on stand this ground. Because you think you thought every side, and you want to show you are considerable.

You also always said arecca is just reference not comparison, but then how your point exist with? Becaue you just randomly thought 98% is great or you farmed too much pp from comfortable maps, or you are just using your experience from other game that you even didn't notice? You could always say you are neutral but this is what is going on from what I see. That's why I use comparison against your comparison even if you think your opinion isn't comparison.
Do you expect me to read your mind or just read the text?

The note lock removal from what I see is a QoL to begin with, if you triggered a note lock(or not in lazer) then it should be a quite big mess, it's just indirectly affecting accuracy and won't happen many times, mostly it is used to prevent failing, not preserve score. And the slider accuracy is even seen as a cause that classic acc would higher than Lazer acc most of the time. What my point is always there, do not be harder after harder.

I do not disregard stable players opinions, I disregard history issue, because they are buffed with a system they know for years, if not with a fixed weight to consider it then nothing can be changed. Or you could go suggest that remove slider accuracy, I don't see which group is larger, no slider acc gang or no miss S gang. Also there is also who against miss requirement and who neutral with it, and Lazer players include stable players just you know. Do not get the wrong idea.

I don't know what gives you the idea that you are trying to consider every side, not even a bit. Even if you said 95%for miss and 98%for no miss I would kinda agree. What is this thing? And called considered every side? By kicking both "can miss but not high as 98%" gang and 95% gang's ass? Meh.

**Reply by Speykious** [reply]
*sigh*

Most unproductive discussion I've had in a while.
I'm not even gonna bother continuing to argue this.

**Reply by LocalGuard039** [reply]
Nor do I

**Comment by LocalGuard039** [comment]


**Reply by Speykious** [reply]
There is a separate discussion thread proposing a visual distinction for FC and non-FC scores, you might want to look at it:
- https://github.com/ppy/osu/discussions/26571

**Comment by Packsolite** [comment]
I've played with quite a lot of beginners and most of them prefered the "no fc" equirement for S.

Personally coming from beat saber, having the grade only based on acc and a seperate indicator for FC would be the best solution imo.
